tracpro/contacts/tests/test_views.py

```diff
@@ -112,6 +96,148 @@ def test_create_with_access(self):
         self.assertEqual(json.loads(response.content),
                          dict(results=[dict(id='kin', text='Kinyarwanda')]))
 
+        # testing form validation
+        # phone number is not numeric
+        response = self.url_post('unicef', url, {
+            'name': "Mo Polls II",
+            'urn_0': "tel",
+            'urn_1': "qwerty",
+            'region': self.region1.pk,
+            'groups': (self.group1.pk, self.group2.pk, self.group3.pk),
+        })
+
+        self.assertEqual(response.status_code, 200)
+        form = response.context['form']
+        self.assertEqual(len(form.errors), 1, form.errors)
+        self.assertFormError(response, 'form', 'urn', ('The string supplied did not seem to be a phone number.'))
+
+        # phone number does not have country code
+        response = self.url_post('unicef', url, {
+            'name': "Mo Polls II",
+            'urn_0': "tel",
+            'urn_1': "2345263746",
+            'region': self.region1.pk,
+            'groups': (self.group1.pk, self.group2.pk, self.group3.pk),
+        })
+
+        self.assertEqual(response.status_code, 200)
+        form = response.context['form']
+        self.assertEqual(len(form.errors), 1, form.errors)
+        self.assertFormError(response, 'form', 'urn', ('Missing or invalid default region.'))
```

Comment:
It seems like the error here should be the one we used earlier, "This phone number has an invalid country code."?  (which seems much clearer than this one)